src/main/java/com/ning/http/client/multipart/AbstractFilePart.java

```diff
@@ -66,7 +66,7 @@ public AbstractFilePart(String name, String contentType, Charset charset, String
 
     protected void visitDispositionHeader(PartVisitor visitor) throws IOException {
         super.visitDispositionHeader(visitor);
-        if (fileName != null) {
+        if (!hasCustomContentDisposition() && fileName != null) {
```

Comment:
why not use StringUtils.isNotBlank(fileName) instead of "fileName != null" to make sure that you don't get an empty string as well.